pkg/providers/apisix/translation/apisix_upstream.go

```diff
@@ -57,35 +55,24 @@ func (t *translator) translateService(namespace, svcName, subset, svcResolveGran
 	return ups, nil
 }
 
-// TODO: Support Port field
 func (t *translator) TranslateApisixUpstreamExternalNodes(au *v2.ApisixUpstream) ([]apisixv1.UpstreamNode, error) {
 	var nodes []apisixv1.UpstreamNode
 	for i, node := range au.Spec.ExternalNodes {
 		if node.Type == v2.ExternalTypeDomain {
-			arr := strings.Split(node.Name, ":")
 
 			weight := translation.DefaultWeight
 			if node.Weight != nil {
 				weight = *node.Weight
 			}
 			n := apisixv1.UpstreamNode{
-				Host:   arr[0],
+				Host:   node.Name,
```

Comment:
this means user only can use `domain`, right?

I think we should add some logic to make sure it is correct and also add test case to cover it.
For example, if the user fills in https://github.com , how should we handle it?

Reply:
Yes, only domain can be used, or more precisely, the regular expression to match host.

```lua
local host_def_pat = "^\\*?[0-9a-zA-Z-._\\[\\]:]+$"
```

ref: https://github.com/apache/apisix/blob/c5fc10d9355a0c177a7532f01c77745ff0639a7f/apisix/schema_def.lua#L40

We can provide user-friendly error messages on ingress.

Reply:
ref to apisix : https://github.com/apache/apisix/blob/c5fc10d9355a0c177a7532f01c77745ff0639a7f/apisix/schema_def.lua#L40
ref to k8s : https://github.com/kubernetes/kubernetes/blob/976a940f4a4e84fe814583848f97b9aafcdb083f/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L205

They define regex differently, but I think k8s's dns is more accurate.

Reply:
I think we can use k8s's rule